dashboard/app/models/school_info.rb

```diff
@@ -100,7 +100,6 @@ def sync_from_schools
       original[:school_name] = school_name
       original[:full_address] = full_address
 
-      school = School.find(school_id) if school.nil?
```

Comment:
This isn't needed because the [school association](https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/app/models/school_info.rb#L47) already binds to `school_id` and will handle that automatically in the DB-persistence case. In the case of built non-persisted objects, this will blow up because the id is fake.